shared/js/background/atb.es6.js

```diff
@@ -173,18 +176,43 @@ const ATB = (() => {
             // - the user wasn't already looking at the app install page
             // - the user hasn't seen the page before
             settings.ready().then(() => {
-                chrome.tabs.query({currentWindow: true, active: true}, function (tabs) {
-                    const domain = (tabs && tabs[0]) ? tabs[0].url : ''
-                    if (ATB.canShowPostInstall(domain)) {
-                        settings.updateSetting('hasSeenPostInstall', true)
-                        let postInstallURL = 'https://duckduckgo.com/app?post=1'
-                        const atb = settings.getSetting('atb')
-                        postInstallURL += atb ? `&atb=${atb}` : ''
-                        chrome.tabs.create({
-                            url: postInstallURL
+                // Special case for the cross product promotion on desktop (cppd) experiment
+                if (settings.getSetting('isMultiStepOnboarding')) {
+                    // We find the tab containing the cross product promotion modal
+                    // and inject a content script that will notify the page that the
+                    // extension was successfully installed
+                    // Note: that we don't know on which window that tab is
+                    chrome.tabs.query({}, (tabs) => {
+                        const tab = tabs.find((tab) => {
+                            const { hostname, searchParams } = new URL(tab.url)
+
+                            return (hostname.includes('duckduckgo.com') && !!searchParams.get('natb'))
                         })
-                    }
-                })
+
+                        const file = 'public/js/content-scripts/onboarding.js'
+                        if (tab) {
```

Comment:
If there's no DDG tab I think we'll want to open a new tab to the homepage so they continue with the multi-step onboarding?

Reply:
apologies if I am missing something but I don't think that we can detect that no ?
If the user closed the tab _before_ installing the extension then `settings.getSetting('isMultiStepOnboarding')` will be `false`  and we can't detect that the install was triggered by the new onboarding workflow.

Reply:
Yup you're right. Not sure what I was thinking